tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc

```diff
@@ -4448,10 +4450,27 @@ class GenericConvertUnsortedSegmentReductionOp : public OpRewritePattern<OpTy> {
     // Broadccast the initial value for reduction. This will become the
     // 'operand' parameter to scatter to for the final scatter op.
     Value init = ConcreteClass::GetInitialValue(data_type.getElementType(),
-                                                op.getLoc(), &rewriter);
-    auto broadcasted_init = rewriter.create<mhlo::BroadcastOp>(
-        op.getLoc(), output_type, init,
-        GetI64ElementsAttr(output_shape, &rewriter));
+                                                loc, &rewriter);
+    ArrayRef<int64_t> data_shape = data_type.getShape();
+    llvm::SmallVector<Value, 4> shape_values;
+    shape_values.reserve(output_shape.size());
+    shape_values.push_back(rewriter.create<mlir::ConstantOp>(
+        loc, builder.getIndexAttr(output_shape[0])));
+    for (int i = segment_ids_rank; i < data_shape.size(); ++i) {
+      if (output_shape[i - segment_ids_rank + 1] == ShapedType::kDynamicSize) {
```

Comment:
How about capturing this as const_dim = output_shape[i - segment_ids_rank + 1] and then using that in the else?